lib/rest-api.php

```diff
@@ -473,3 +473,20 @@ function gutenberg_filter_user_collection_parameters( $query_params ) {
 	return $query_params;
 }
 add_filter( 'rest_user_collection_params', 'gutenberg_filter_user_collection_parameters' );
+
+/**
+ * Silence PHP Warnings and Errors in JSON requests
+ *
+ * @todo This is a temporary measure until errors are properly silenced in REST API responses in core
+ *
+ * @see https://core.trac.wordpress.org/ticket/44534
+ */
+function gutenberg_silence_rest_errors() {
+
+	if ( ! empty( $_SERVER['CONTENT_TYPE'] ) && 'application/json' === $_SERVER['CONTENT_TYPE'] ) {
```

Comment:
As it turns out, we may also need to inspect `HTTP_ACCEPT` in addition to `CONTENT_TYPE`: https://core.trac.wordpress.org/ticket/44534#comment:18